neo/Ledger/TrimmedBlock.cs

```diff
@@ -57,7 +57,7 @@ public Header Header
         public override void Deserialize(BinaryReader reader)
         {
             base.Deserialize(reader);
-            Hashes = reader.ReadSerializableArray<UInt256>();
+            Hashes = reader.ReadSerializableArray<UInt256>(Block.MaxTransactionsPerBlock);
```

Comment:
The same here, I did not get this change.

Reply:
In case a malicious node serializes with a larger than it should size this stops the server from eating tons of memory...